runtime/integration-tests/src/xcm/polkadot/tests/currency_id_convert.rs

```diff
@@ -131,37 +131,15 @@ fn convert_cfg_xcm_v2() {
 	});
 }
 
-/// Verify that Tranche tokens are not handled by the CurrencyIdConvert
-/// since we don't allow Tranche tokens to be transferable through XCM.
+/// Verify that a registered token that is NOT XCM transferable is filtered out
+/// by CurrencyIdConvert as expected.
 #[test]
-fn convert_tranche() {
-	// We don't yet know the pools pallet index on the Centrifuge runtime
-	const MOCK_POOLS_PALLET_INDEX: u8 = 42;
-	let tranche_currency = CurrencyId::Tranche(401, [0; 16]);
-	let tranche_id =
-		WeakBoundedVec::<u8, ConstU32<32>>::force_from(tranche_currency.encode(), None);
-	let tranche_multilocation = MultiLocation {
-		parents: 1,
-		interior: X3(
-			Parachain(parachains::polkadot::centrifuge::ID),
-			PalletInstance(MOCK_POOLS_PALLET_INDEX),
-			GeneralKey {
-				length: tranche_id.len() as u8,
-				data: vec_to_fixed_array(tranche_id.to_vec()),
-			},
-		),
-	};
-
+fn convert_no_xcm_token() {
 	Centrifuge::execute_with(|| {
-		assert_eq!(
-			<CurrencyIdConvert as C1<_, _>>::convert(tranche_multilocation),
-			Err(tranche_multilocation),
-		);
-	});
+		register_no_xcm_token();
 
-	Centrifuge::execute_with(|| {
 		assert_eq!(
-			<CurrencyIdConvert as C2<_, _>>::convert(tranche_currency),
+			<CurrencyIdConvert as C2<_, _>>::convert(NO_XCM_ASSET_ID),
 			None
 		)
 	});
```

Comment:
Yes! Note how we registered the "no xcm token" and then verify that the conversion fails; the conversion is a key step in processing an XCM transfer since it tries to convert `CurrencyId` to `MultiLocation` for an outbound transfer and the reverse in an incoming one; we don't need to test the incoming case because we can't receive tokens that were not sent out in the first place.

I could write an explicit test showing that xtokens.transfer fails for a "no xcm token", would you like to see that? I didn't do that to avoid redundancy but let me know how you see it

Reply:
> we don't need to test the incoming case because we can't receive tokens that were not sent out in the first place.

But we can receive foreign assets that never hit our chain before, right?